dspace-api/src/main/java/org/dspace/uclouvain/profileIngester/ProfileIngesterCLI.java

```diff
@@ -0,0 +1,194 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(ProfileIngesterCLI.class);
+
+    // EXTERNAL SERVICES
+    protected EPersonService ePersonService;
+    protected ResearcherProfileService researcherProfileService;
+    protected ConfigurationService configService;
+    protected PersonEventConnector personEventConnector;
+
+    // COMMON CLICommand METHODS----------------------------------
+    @Override
+    protected void buildOptions() {
+        serviceOptions.addOption(OPT_PERSON);
+        serviceOptions.addOption(OPT_ERROR_QUEUE_NAME);
+        infoOptions.addOption(OPT_HELP);
+    }
+
+    @Override
+    protected String getUsageDescription() {
+        return USAGE_DESCRIPTION;
+    }
+
+    // METHODS----------------------------------------------------
+    public static void main(String[] arguments) throws Exception {
+        ProfileIngesterCLI profileIngester = new ProfileIngesterCLI();
+        CommandLine cli = profileIngester.validateCLIArgument(arguments);
+
+        PersonEventConnector connector = new PersonEventConnector();
+
+        String epersonEmail = cli.getOptionValue("e");
+        String errorQueueName = cli.getOptionValue("eq", connector.getQueueName() + "_error");
+        while (true) {
+            try {
+                profileIngester.ingestEvent(epersonEmail, connector, errorQueueName);
+            } catch (InterruptedException ie) {
+                logger.info("Execution interrupted manually");
+                Thread.currentThread().interrupt();
+                break;
+            } catch (Exception e) {
+                logger.error(e.getMessage(), e);
+            }
+        }
+    }
+
+    ProfileIngesterCLI() throws IOException, TimeoutException {
+        ePersonService = EPersonServiceFactory.getInstance().getEPersonService();
+        researcherProfileService = new DSpace().getSingletonService(ResearcherProfileService.class);
+    }
+
+    /**
+     * Launches the cli ingest process: wait for events in the rabbit queue and do the required action.
+     * @param email The user email to use an initiator for the ingest process actions.
+     * @param connector The RabbitMQ connector to use to get event messages from queue.
+     * @param errorQueueName The queue name to post potential error message to.
+     * @throws Exception
+     */
+    private void ingestEvent(String email, PersonEventConnector connector, String errorQueueName) throws Exception {
+        // Set a random uuid to this thread context to identify this job in the logs.
+        String jobId = UUID.randomUUID().toString();
+        ThreadContext.put("jobId", jobId);
+
+        // Get the eperson to use as initiator.
+        Context context = new Context();
+        EPerson ePerson = ePersonService.findByEmail(context, email);
+        if (ePerson == null) {
+            throw new UserNotFoundException(email);
+        }
```

Comment:
Remove from the "loop" function to be placed into the "main" function to be executed once